### PR TITLE
Allow maximum age of csrf cookie to be configured

### PR DESCRIPTION
**Pull request by sedran**
Allows maxAge of the generated cookie by CookieCsrfTokenRepository
to be configurable.

Prior to this commit, maximum age was set with a value of -1.

After this commit, it will be configured by the user with an either
positive or negative value. If the user does not provide a value,
it will be set -1.

An IllegalArgumentException will be thrown when
this value is set to zero.

Closes gh-9195

### CODE REVIEW

File: web/src/main/java/org/springframework/security/web/csrf/CookieCsrfTokenRepository.java [hunk 1]
```diff
@@ -192,4 +194,30 @@ public void setSecure(Boolean secure) {
...
+	 * <p>
+	 * A zero value causes the cookie to be deleted immediately therefore it is not a
+	 * valid value and in that case an {@link IllegalArgumentException} will be thrown.
+	 * @param cookieMaxAge an integer specifying the maximum age of the cookie in seconds;
+	 * if negative, means the cookie is not stored; if zero, the method throws an
+	 * {@link IllegalArgumentException}
+	 * @since 5.5
+	 */
+	public void setCookieMaxAge(int cookieMaxAge) {
+		Assert.isTrue(cookieMaxAge != 0, "cookieMaxAge is not zero");
```
Comment (rwinch) [inline web/src/main/java/org/springframework/security/web/csrf/CookieCsrfTokenRepository.java:219]: Can you please change this to be cookieMaxAge cannot be zero? As it is, the error the user will see if it is `0` would state it is not zero which would be confusing.

```suggestion
		Assert.isTrue(cookieMaxAge != 0, "cookieMaxAge cannot be zero");
```

**Review by rwinch — CHANGES_REQUESTED** [review]
Thanks for the PR! I've provided feedback inline.

File: web/src/main/java/org/springframework/security/web/csrf/CookieCsrfTokenRepository.java (hunk 1 above)
Comment (sedran) [reply web/src/main/java/org/springframework/security/web/csrf/CookieCsrfTokenRepository.java:219]: Thank you for your feedback. In this class there are other assertions written in this way. I've tried to follow the convention in the class. Can I change the other assertions in the same file in order to be consistent?

**Comment by sedran** [comment]
The automated build fails because of a dependency error from spring maven repository. How can we solve that?

```
> Could not HEAD 'https://repo.spring.io/libs-snapshot/org/infinispan/infinispan-bom/11.0.5.Final/infinispan-bom-11.0.5.Final.pom'. 
Received status code 401 from server: Unauthorized
```

**Comment by jzheaux** [comment]
Sorry for the issue you experienced with the build, @sedran - it was due to an infrastructural change with Spring's Maven repositories. 

If you rebase, then the error should be addressed.

**Comment by sedran** [comment]
> Sorry for the issue you experienced with the build, @sedran - it was due to an infrastructural change with Spring's Maven repositories.
> 
> If you rebase, then the error should be addressed.

I have amend/force-pushed my branch to trigger auto build as soon as I see the CI status of the project is green. 
Now I can see "All checks have passed" text and there is no conflict. I also implemented the requested changes by @rwinch .

Is there anything I can do for this PR to be merged? Please let me know.

Thanks.

**Review by rwinch — APPROVED** [review]
Thanks for the updates and the ping. The changes look good to me

**Pull request merged**